### PR TITLE
perf: improve booking confirm authorization

### PR DESCRIPTION
**Pull request by anonrig**
Previous implementation was wrong and unperformant. This pull request fixes the following bug, and reduces application layer data and moves some comparison to the database layer.

```
eventType.users.find((user) => user.id === user.id)
```

(It was always returning truthy)

### CODE REVIEW

**Review by roae — APPROVED** [review]
Looks good! 🙏

**Review by zomars — APPROVED** [review]
Thank you for your contribution 🙏

File: packages/trpc/server/routers/viewer/bookings.tsx [hunk 1]
```diff
@@ -781,30 +781,22 @@ export const bookingsRouter = router({
...
 
     const authorized = async () => {
-      // if the organizer
-      if (booking.userId === user.id) {
-        return true;
-      }
       const eventType = await prisma.eventType.findUnique({
         where: {
           id: booking.eventTypeId || undefined,
+          schedulingType: SchedulingType.COLLECTIVE,
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/bookings.tsx:787]: We have a type error here:

image

Comment (zomars) [reply packages/trpc/server/routers/viewer/bookings.tsx:787]: Pushed a fix

**Pull request merged**